x-pack/solutions/security/plugins/security_solution/public/timelines/store/middlewares/timeline_save.ts

```diff
@@ -51,124 +53,159 @@ function isSaveTimelineAction(action: Action): action is ReturnType<typeof saveT
-      store.dispatch(startTimelineSaving({ id: localTimelineId }));
+    const { dataViewId: experimentalDataViewId } = sourcererAdapterSelector(
+      SourcererScopeName.timeline
+    )(storeState);
 
-      try {
-        const response = await (action.payload.saveAsNew && timeline.id
-          ? copyTimeline({
-              timelineId,
-              timeline: {
-                ...convertTimelineAsInput(timeline, timelineTimeRange),
-                dataViewId: selectedDataViewIdSourcerer,
-                indexNames: selectedPatterns,
-                templateTimelineId,
-                templateTimelineVersion,
-              },
-              savedSearch: timeline.savedSearch,
-            })
-          : persistTimeline({
-              timelineId,
-              version: timelineVersion,
-              timeline: {
-                ...convertTimelineAsInput(timeline, timelineTimeRange),
-                dataViewId: selectedDataViewIdSourcerer,
-                indexNames: selectedPatterns,
-                templateTimelineId,
-                templateTimelineVersion,
-              },
-              savedSearch: timeline.savedSearch,
-            }));
+    const experimentalIsDataViewEnabled =
+      storeState.app.enableExperimental.newDataViewPickerEnabled;
 
-        if (isTimelineErrorResponse(response)) {
-          const error = getErrorFromResponse(response);
-          switch (error?.errorCode) {
-            case 403:
-              store.dispatch(showCallOutUnauthorizedMsg());
-              break;
-            // conflict
-            case 409:
-              kibana.notifications.toasts.addDanger({
-                title: i18n.TIMELINE_VERSION_CONFLICT_TITLE,
-                text: i18n.TIMELINE_VERSION_CONFLICT_DESCRIPTION,
-              });
-              break;
-            default:
-              kibana.notifications.toasts.addDanger({
-                title: i18n.UPDATE_TIMELINE_ERROR_TITLE,
-                text: error?.message ?? i18n.UPDATE_TIMELINE_ERROR_TEXT,
-              });
-          }
-          return;
-        }
+    let experimentalSelectedPatterns: string[] = [];
+
+    // NOTE: remove eslint override above after the experimental picker is stabilized
+    if (experimentalIsDataViewEnabled && experimentalDataViewId) {
+      const plugins = await kibana.plugins.onStart<{ dataViews: DataViewsPublicPluginStart }>(
+        'dataViews'
+      );
+
+      if (plugins.dataViews.found) {
+        const experimentalDataView = await plugins.dataViews.contract.get(experimentalDataViewId);
 
-        if (response == null) {
-          kibana.notifications.toasts.addDanger({
-            title: i18n.UPDATE_TIMELINE_ERROR_TITLE,
-            text: i18n.UPDATE_TIMELINE_ERROR_TEXT,
-          });
-          return;
+        if (!experimentalDataView.isPersisted()) {
+          // TODO: better wording here and a lock on the ui side?
+          throw new Error('Persting timelines with adhoc data views is not allowed');
```

Comment:
We can either do this, or auto-persist the DV on timeline save? I'm sure users don't want additional tasks to block their timeline save..

Reply:
Also, we should make this a toast if we want to notify users as it only shows up in the console now

Reply:
added a toast, will think about solving it better in the future, thanks